tests/unittests/test_datasource/test_common.py

```diff
@@ -95,5 +97,12 @@ def test_data_sources_have_valid_network_config_sources(self):
                 self.assertTrue(hasattr(sources.NetworkConfigSource, cfg_src),
                                 fail_msg)
 
+    def test_expected_dsname_defined(self):
+        for ds in DEFAULT_LOCAL + DEFAULT_NETWORK:
+            fail_msg = (
+                '{} has an invalid / missing dsname property'.format(str(ds))
```

Comment:
If we're going to override the default message, I think we should include at least `ds.dsname` in the message.